app/models/patient.rb

```diff
@@ -34,31 +34,36 @@ class Patient < ApplicationRecord
   }
 
   %i[address_state
-     ethnicity
      monitored_address_state
+     foreign_monitored_address_state
+     additional_planned_travel_destination_state
+     ethnicity
      preferred_contact_method
      preferred_contact_time
      sex
      primary_telephone_type
-     secondary_telephone_type].each do |enum_field|
-    validates enum_field, on: :api, inclusion: {
+     secondary_telephone_type
+     additional_planned_travel_type
+     case_status].each do |enum_field|
+    validates enum_field, on: %i[api import], inclusion: {
       in: VALID_PATIENT_ENUMS[enum_field],
-      message: "is not an acceptable value, acceptable values are: '#{VALID_PATIENT_ENUMS[enum_field].join("', '")}'"
-    }, allow_blank: true
+      message: "is not an acceptable value, acceptable values are: '#{VALID_PATIENT_ENUMS[enum_field].reject(&:blank?).join("', '")}'"
+    }
   end
 
   %i[primary_telephone
      secondary_telephone].each do |phone_field|
-    validates phone_field, on: :api, phone_number: true
+    validates phone_field, on: %i[api import], phone_number: true
   end
 
   %i[date_of_birth
```

Comment:
would it make sense to define these arrays in ValidationHelper or other helper? some of the export code (https://github.com/SaraAlert/SaraAlert/blob/master/app/lib/import_export.rb#L121) might be able to benefit from refactoring these lists of fields by data types into one place

Reply:
I've looked a bit closer and I'm not entirely sure we can overlap the validation lists with the lists of data types, since the data type is not always tied to how the field is validated. For example, that list of `string`s includes fields that are validated as `enum` types, fields that are validated as `email`, and fields that are not validated at all, since they can just be any string. There does exist a list in `ValidationHelper` that lists out the specific types of validation done on each field: https://github.com/SaraAlert/SaraAlert/blob/064dd5d6ad7c8d92e7d4a6197f4c9a16d04a8fb0/app/helpers/validation_helper.rb.

Reply:
no problem, we can just leave it as is then, thanks for looking into it though!